azkaban-common/src/main/java/azkaban/executor/selector/ExecutorComparator.java

```diff
@@ -0,0 +1,307 @@
+    return comparatorCreatorRepository.keySet();
+  }
+
+  // factor comparator names
+  private static final String REMAININGFLOWSIZE_COMPARATOR_NAME = "RemainingFlowSize";
+  private static final String MEMORY_COMPARATOR_NAME = "Memory";
+  private static final String REMAININGTMPSIZE_COMPARATOR_NAME = "RemainingTmpSize";
+  private static final String PRIORITY_COMPARATOR_NAME = "Priority";
+  private static final String LSTDISPATCHED_COMPARATOR_NAME = "LastDispatched";
+  private static final String CPUUSAGE_COMPARATOR_NAME = "CpuUsage";
+
+  /**
+   * static initializer of the class.
+   * We will build the filter repository here.
+   * when a new comparator is added, please do remember to register it here.
+   * */
+  static {
+    comparatorCreatorRepository = new HashMap<String, ComparatorCreator>();
+
+    // register the creator for remaining flow size comparator.
+    comparatorCreatorRepository.put(REMAININGFLOWSIZE_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getRemainingFlowSizeComparator(weight); }});
+
+    // register the creator for memory comparator.
+    comparatorCreatorRepository.put(MEMORY_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getMemoryComparator(weight); }});
+
+    // register the creator for priority comparator.
+    comparatorCreatorRepository.put(PRIORITY_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getPriorityComparator(weight); }});
+
+    // register the creator for remaining TMP size comparator.
+    comparatorCreatorRepository.put(PRIORITY_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getRemainingTmpSizeComparator(weight); }});
+
+    // register the creator for last dispatched time comparator.
+    comparatorCreatorRepository.put(LSTDISPATCHED_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getLstDispatchedTimeComparator(weight); }});
+
+    // register the creator for CPU Usage comparator.
+    comparatorCreatorRepository.put(CPUUSAGE_COMPARATOR_NAME, new ComparatorCreator(){
+      @Override public FactorComparator<Executor> create(int weight) { return getCpuUsageComparator(weight); }});
+  }
+
+
+  /**
+   * constructor of the ExecutorComparator.
+   * @param comparatorList   the list of comparator, plus its weight information to be registered,
+   *  the parameter must be a not-empty and valid list object.
+   * */
+  public ExecutorComparator(Map<String,Integer> comparatorList) {
+    if (null == comparatorList|| comparatorList.size() == 0){
+      logger.error("failed to initialize executor comparator as the passed comparator list is invalid or empty.");
+      throw new IllegalArgumentException("filterList");
```

Comment:
I think you can move the message into the IllegalArgumentException exception.  The log statement is optional.